staging/src/k8s.io/dynamic-resource-allocation/structured/internal/experimental/allocator_experimental.go

```diff
@@ -204,21 +204,9 @@ func (a *Allocator) Allocate(ctx context.Context, node *v1.Node, claims []*resou
 
 			// Error out if the consumableCapacity feature is not enabled
 			// and the request contains capacity requests.
-			if !a.features.ConsumableCapacity {
-				containsCapacityRequest := false
-				if request.Exactly != nil && request.Exactly.Capacity != nil {
-					containsCapacityRequest = true
-				}
-				for _, request := range request.FirstAvailable {
-					if request.Capacity != nil {
-						containsCapacityRequest = true
-						break
-					}
-				}
-				if containsCapacityRequest {
-					return nil, fmt.Errorf("claim %s, request %s: has capacity requests, but the DRAConsumableCapacity feature is disabled",
-						klog.KObj(claim), request.Name)
-				}
+			if !a.features.ConsumableCapacity && request.Exactly != nil && request.Exactly.Capacity != nil {
```

Comment:
The previous loop over subrequests is a redundant check. The subrequest will be also checked later in hasSubRequests condition. We may decrease LOC with the current change or remove the check below and keep this loop for quick error out.

https://github.com/kubernetes/kubernetes/blob/faad6b2c30d7f52080185b32ce9a0b0e0916150a/staging/src/k8s.io/dynamic-resource-allocation/structured/internal/experimental/allocator_experimental.go#L233-L239